### PR TITLE
Adding SnapSettingsCard ui component

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Description
Part of: #12574

Explanation: Adding `SnapSettingsCard` ui component

**TO DO**
- [x] Unit tests
- [x] Move `Card` components to a separate PR for `develop` #12666
- [x] Move `Pill` updates to separate PR #12668 
- [x] Move `SnapSettingCard` to `flask/` subfolder
- [x] Confirm `ComponentName`
- [x] ~~Figure out how to display icons with url~~
- [x] ~~Break into smaller sub components (needed?)~~
- [x] Resolve card body overflow ellipsis without using web-kit [**web-kit was used since there's browser support for it**]
- [x] Add localization (maybe do on page level to keep component dumb)

Manual testing steps:  
  - Go to latest [storybook build ](https://397221-42009758-gh.circle-artifacts.com/0/storybook/index.html?path=/story/ui-components-app-flask-snap-settings-card-snap-settings-card-stories-js--default-story)
  - Search for "snapsettingcard"
  - Test controls
  
  ## Images

Screen Shot 2021-11-23 at 4 08 47 PM
Screen Shot 2021-11-23 at 4 08 55 PM

### CODE REVIEW

File: ui/components/app/app-components.scss [hunk 1]
```diff
@@ -46,4 +47,5 @@
 @import 'transaction-total-banner/index';
 @import 'wallet-overview/index';
 @import 'whats-new-popup/index';
-@import 'loading-network-screen/index'
+
+/** Please import your files in alphabetical order **/
```
Comment (georgewrmarshall) [inline ui/components/app/app-components.scss:51]: Arranging alphabetically

File: ui/components/ui/circle-icon/circle-icon.stories.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ export const basicCircleIcon = () => (
     border="1px solid"
     borderColor="black"
     background="white"
-    iconSize="42px"
+    iconSize="20px"
+    size="46px"
     iconSource="images/eth_logo.svg"
   />
 );
```
Comment (georgewrmarshall) [inline ui/components/ui/circle-icon/circle-icon.stories.js:18]: Revert changes

**Review by hmalik88 — COMMENTED** [review]
I had a couple questions/comments:

1. This one might be silly 😅, but for the storybook file, I saw that you passed in `AST.png` for the `iconUrl` prop, which eventually ends up in an `img` tag in the `IconWithFallback` component. Where does that `.png` file actually live?

2. I was going to mention localization, but then saw you mention it in the checklist as well. What are we using for localization? Just realized the chip and card component did not have localization either (will comment there as a reminder to myself if you don't put it in)

3. Not 100% sure if we should break into subcomponents, my gut feeling is leaning towards yes? I guess it depends on if we perceive this component getting more complex in the future...also it would look a bit cleaner with less LOC. I'm open to hearing opinions about this, though.

4. I'm not sure if we need the `useCallback` here, from my understanding it's primarily used to pass callbacks to children components that rely on reference quality to prevent re-renders. I think the `useEffect` alone is fine here, the components that rely on render will only ever re-render if `chipStatus` changes, unless I'm missing something?

5. What is the line right above the snap name? I didn't see it as a part of the design.

Screen Shot 2021-11-12 at 6 30 58 PM

6. Just to confirm the remaining work is: unit tests, card body overflow and localization?

**Comment by georgewrmarshall** [comment]
@hmalik88 Answers to your questions below
1. This image lives in `.storybook/images` not sure why it's not showing up have added it to TODO list
2. Yes, not sure where we stand on localization have commented on chip. I would tend to lean towards doing localization at the level the text data is coming in. So may need some here for statuses but all the rest should be at page level? Maybe check with @rekmarks 
3. I'm on the fence here too. I know breaking components up into smaller may have an effect on performance. It's all about finding that balance. I feel like I'll leave it for now incase anything changes. I can do some refactoring when I get back if you think it's valuable.
4. I used useCallback because the linter was complaining. But still not sure. I need to scrub up on my hook knowledge
5. Yes so! The line should be removed once Card is merged to develop and you rebase. Padding of the card, and the pill background should also update once those PRs are in too. You should not have to make any changes to the css code here

**Comment by hmalik88** [comment]
> @hmalik88 Answers to your questions below
> 
> 1. This image lives in `.storybook/images` not sure why it's not showing up have added it to TODO list
> 2. Yes, not sure where we stand on localization have commented on chip. I would tend to lean towards doing localization at the level the text data is coming in. So may need some here for statuses but all the rest should be at page level? Maybe check with @rekmarks
> 3. I'm on the fence here too. I know breaking components up into smaller may have an effect on performance. It's all about finding that balance. I feel like I'll leave it for now incase anything changes. I can do some refactoring when I get back if you think it's valuable.
> 4. I used useCallback because the linter was complaining. But still not sure. I need to scrub up on my hook knowledge
> 5. Yes so! The line should be removed once Card is merged to develop and you rebase. Padding of the card, and the pill background should also update once those PRs are in too. You should not have to make any changes to the css code here

Great, thanks! I'll take a look at the stuff you don't get to.

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+  status,
+  className,
+  cardProps,
+  toggleButtonProps,
+  buttonProps,
+  chipProps,
+}) => {
+  const [chipStatus, setChipStatus] = useState({
+    label: STATUSES.INSTALLING,
+    indicatorColor: STATUS_COLORS[STATUSES.INSTALLING],
```
Comment (darkwing) [inline ui/components/app/flask/snap-settings-card/snap-settings-card.js:56]: Looking at how `indicatorColor` is used, it always mirrors its status.  Do we need the separate key/value?

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        </Box>
+      </Box>
+      <Typography
+        variant={TYPOGRAPHY.Paragraph}
+        color={COLORS.UI4}
+        fontWeight={FONT_WEIGHT.NORMAL}
+        className="snap-settings-card__body"
+        boxProps={{
+          marginBottom: 4,
+          marginTop: 0,
```
Comment (ryanml) [inline ui/components/app/flask/snap-settings-card/snap-settings-card.js:144]: Can we remove `marginTop` and `margin` here and just define `marginBottom`?

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js (hunk 3 above)
Comment (hmalik88) [reply ui/components/app/flask/snap-settings-card/snap-settings-card.js]: Good catch, I overlooked this in my initial review. Thanks!

File: ui/components/ui/icon-border/icon-border.scss [hunk 5]
```diff
@@ -1,8 +1,9 @@
...
   border-radius: 50%;
-  border: 1px solid #f2f3f4;
-  background: #ececf0;
+  border: 1px solid $ui-1;
+  background: $ui-1;
   display: flex;
   justify-content: center;
   align-items: center;
+  overflow: hidden;
 }
```
Comment (georgewrmarshall) [inline ui/components/ui/icon-border/icon-border.scss:9]: Checked instances of the `<SiteIcon>` component and visual changes should be minimal. There is a hex value above that isn't apart of our official color palette.

Screen Shot 2021-11-23 at 3 51 25 PM
Screen Shot 2021-11-23 at 3 52 20 PM
Screen Shot 2021-11-23 at 3 53 28 PM

Comment (georgewrmarshall) [reply ui/components/ui/icon-border/icon-border.scss]: Resolved in #12797

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+            alignItems={ALIGN_ITEMS.CENTER}
+            marginBottom={4}
+          >
+            <Box>
+              <Button
+                className="snap-settings-card__button"
+                onClick={onClick}
+                {...buttonProps}
+              >
+                See Details
```
Comment (darkwing) [inline ui/components/app/flask/snap-settings-card/snap-settings-card.js:148]: Needs localization.

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+                boxProps={{
+                  margin: [0, 0],
+                }}
+                color={COLORS.UI3}
+                variant={TYPOGRAPHY.H8}
+                fontWeight={FONT_WEIGHT.NORMAL}
+                tag="span"
+                className="snap-settings-card__date-added"
+              >
+                {dateAdded && `Added on`} {dateAdded} {dateAdded && `from`}{' '}
```
Comment (darkwing) [inline ui/components/app/flask/snap-settings-card/snap-settings-card.js:184]: Needs localization

**Review by darkwing — CHANGES_REQUESTED** [review]

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js (hunk 6 above)
Comment (georgewrmarshall) [reply ui/components/app/flask/snap-settings-card/snap-settings-card.js]: Added localization

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js (hunk 7 above)
Comment (georgewrmarshall) [reply ui/components/app/flask/snap-settings-card/snap-settings-card.js]: Added localization

**Review by hmalik88 — DISMISSED** [review]

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+import {
+  COLORS,
+  TYPOGRAPHY,
+  FONT_WEIGHT,
+  ALIGN_ITEMS,
+  DISPLAY,
+  TEXT_ALIGN,
+} from '../../../../helpers/constants/design-system';
+
+const STATUSES = {
```
Comment (shanejonas) [inline ui/components/app/flask/snap-settings-card/snap-settings-card.js:26]: we could pull the enum type exported from snap controller: https://github.com/MetaMask/snaps-skunkworks/blob/main/packages/controllers/src/snaps/SnapController.ts#L156-L161

or pull that out into a more common `@metamask/snap-types` package

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+      }
+      default: {
+        setChipStatus(STATUSES.INSTALLING);
+      }
+    }
+  }, [status]);
+
+  useEffect(() => {
+    handleStatus(status);
+  }, [handleStatus, status]);
```
Comment (shanejonas) [inline ui/components/app/flask/snap-settings-card/snap-settings-card.js:86]: I think we can pass `status` to the chip and avoid trying and keep them in sync.

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js (hunk 8 above)
Comment (hmalik88) [reply ui/components/app/flask/snap-settings-card/snap-settings-card.js:26]: but that's typescript though no? how would we use that enum with regular old react?

File: ui/components/app/flask/snap-settings-card/snap-settings-card.js (hunk 9 above)
Comment (hmalik88) [reply ui/components/app/flask/snap-settings-card/snap-settings-card.js:86]: Hmm, I think I agree with this -- seems simpler...

**Review by shanejonas — DISMISSED** [review]

**Review by shanejonas — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**